app/main.py

```diff
@@ -2,0 +1,32 @@
+class Animal:
+    alive = []
+
+    def __init__(self, name: str,
+                 health: int = 100,
+                 hidden: bool = False) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
```

Comment:
Appending `self` to the class variable `alive` in the constructor means that every instance of `Animal` or its subclasses will be added to this list. Ensure that instances are properly removed from this list when they are no longer needed to avoid memory leaks or incorrect state.